### PR TITLE
ENH: match geopandas sjoin and clip API

### LINKED ISSUE #127: clip calls wrong clip function
Hey all.

I am loving the work on this project!
I've stumbled upon a bug and wanted you to let you know about it.
I don't know if it is actually worth "fixing".
If one calls `.clip` on a geopandas dask dataframe object, the dask clip function (clipping values between `min/max` is called)

To reproduce:

```python
import geopandas
import dask_geopandas

path_to_data = geopandas.datasets.get_path("nybb")
gdf = geopandas.read_file(path_to_data)
ddf = dask_geopandas.from_geopandas(gdf, npartitions=2)
ddf.clip(gdf.geometry[0])
```

### PR DESCRIPTION
**Pull request by martinfleis**
Closes #127, xref #130

- add `clip` as a GeoDataFrame method
- support GeoSeries in `clip` and add it the method
- add `sjoin` as a GeoDataFrame method
- change `op` to `predicate` in `sjoin`

### CODE REVIEW

File: dask_geopandas/sjoin.py [hunk 1]
```diff
@@ -38,6 +40,14 @@ def sjoin(left, right, how="inner", op="intersects"):
...
     of the left and right GeoDataFrame.
     """
+    if "op" in kwargs:
+        predicate = kwargs.pop("op")
+        deprecation_message = (
+            "The `op` parameter is deprecated and will be removed"
+            " in a future release. Please use the `predicate` parameter"
+            " instead."
+        )
+        warnings.warn(deprecation_message, FutureWarning)
```
Comment (jorisvandenbossche) [inline dask_geopandas/sjoin.py:50]:
```suggestion
        warnings.warn(deprecation_message, FutureWarning, stacklevel=2)
```

File: dask_geopandas/sjoin.py [hunk 2]
```diff
@@ -46,8 +56,8 @@ def sjoin(left, right, how="inner", op="intersects"):
     if isinstance(right, geopandas.GeoDataFrame):
         right = from_geopandas(right, npartitions=1)
 
-    name = "sjoin-" + tokenize(left, right, how, op)
-    meta = geopandas.sjoin(left._meta, right._meta, how=how, op=op)
+    name = "sjoin-" + tokenize(left, right, how, predicate)
+    meta = geopandas.sjoin(left._meta, right._meta, how=how, predicate=predicate)
```
Comment (jorisvandenbossche) [inline dask_geopandas/sjoin.py:60]: This essentially makes us require geopandas>=0.10 I think? (which is fine for me, but maybe add the lower bounds in setup.py)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

File: dask_geopandas/sjoin.py (hunk 2 above)
Comment (martinfleis) [reply dask_geopandas/sjoin.py:60]: Ah, true. I'll change the requirements. I think that there's no harm having versions of geopandas and dask-geopandas closely coupled, i.e. we can always require latest geopandas with the latest dask-geopandas.

Comment (jorisvandenbossche) [reply dask_geopandas/sjoin.py:60]: Yes, and especially now for releasing the first non-alpha release, it's fine to require the latest geopandas release.

**Pull request merged**